core-api/src/main/java/com/optimizely/ab/Optimizely.java

```diff
@@ -1468,6 +1468,15 @@ public ODPManager getODPManager() {
         return odpManager;
     }
 
+
+    /**
+     * Send an event to the ODP server.
+     *
+     * @param type the event type (default = "fullstack").
+     * @param action the event action name.
+     * @param identifiers a dictionary for identifiers. The caller must provide at least one key-value pair.
+     * @param data a dictionary for associated data. The default event data will be added to this data before sending to the ODP server.
+     */
     public void sendODPEvent(@Nullable String type, @Nonnull String action, @Nullable Map<String, String> identifiers, @Nullable Map<String, Object> data) {
```

Comment:
It think we should change annotation of identifiers to Nonnull, if we are expecting that caller must provide at least one key value pair.

Reply:
@mnoman09 It turns out that this nullable option is needed to support Android (which sets non-empty common identifiers when configuring the java core). I added this clarification into the doc string. 
Another option was to add "vuid" identifier in android-sdk before passing down to java-sdk, but it was not clean since some implicit events ("identifyUser") are created in the java core.

Reply:
But we are going to make this change in android as well right? Or android had to pass identifiers as null?

Reply:
Android (and other client-sdks) customers are allowed to call with empty since SDK is supposed to add at least one identifiers (vuid).

Reply:
Sounds good.